bigquery_etl/metadata/validate_metadata.py

```diff
@@ -27,6 +32,55 @@ def validate_public_data(metadata, path):
     return is_valid
 
 
+def validate_change_control(
+    file_path,
+    metadata,
+    codeowners_file,
+    project_id="moz-fx-data-shared-prod",
+    sql_dir="sql",
+):
+    """Verify that a query is correctly setup for change control."""
+    path_to_add = file_path.partition(f"{project_id}/")[2]
+    path_in_codeowners = os.path.join(sql_dir, project_id, path_to_add)
+    has_change_control = CHANGE_CONTROL_LABEL in metadata.labels
+
+    if has_change_control:
+        # This label requires to have at least one owner in the metadata file.
+        # And for any of the owners, at least one entry in the CODEOWNERS file.
+        # The owners can be emails or GitHub identities e.g. mozilla/team_name.
+        if len(metadata.owners) == 0:
+            click.echo(
+                click.style(
+                    f"The metadata for {file_path} has the label"
+                    f" change_controlled but it's missing code owners."
+                )
+            )
+            return
+
+        with open(codeowners_file, "r") as owners_file:
+            content = owners_file.readlines()
+        occurrences = 0
+        rows_expected = []
+        for meta_owner in metadata.owners:
+            if "@" not in meta_owner:
+                meta_owner = f"@{meta_owner}"
+            row_to_search_for = f"/{path_in_codeowners} {meta_owner}"
+            rows_expected.append(row_to_search_for)
+            for line in content:
+                if row_to_search_for == line.rstrip("\n"):
```

Comment:
I'm still somewhat confused about the use of `occurrences` here, which will always be `0` or `1`, since this check appears to require that there be at least one line matching the format `row_to_search_for`. That value is generated for each metadata owner, one at a time, so per the string comparison strictly requires that there exist a `CODEOWNERS` line containing only a single `CODEOWNER` from among those specified in metadata. Therefore I would expect the following to fail the check:

```
/{path_in_codeowners} {meta_owner} {other_meta_owner}
```

which is particularly confusing because  the error message that is emitted presumably looks like:

`The metadata includes label change_controlled, which requires CODEOWNERS to include at least one owner's record: [{meta_owner}, {other_meta_owner}]`

The requirements in the error message are met by my example (`two` relevant CODEOWNERS is at least `one`). The most trivial way to deal with this is to change the wording and specification to require `exactly one owner's record` instead of `at least one owner's record`, to match what the check implements, but I think we should relax the parsing of the CODEOWNERS line to match the natural language specification and error message.

Related but separate, due to [precedence semantics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax), a CODEOWNERS file might pass the CI check but fail to enforce the expected change control via the following:

```
/{path_in_codeowners} @meta-owner
/{path_in_codeowners} @user-not-in-codeowners
```

This should be caught by a human reviewing the PR so it's perhaps not worth explicitly verifying in CI, but given that the we're implementing these checks to be very specific about change control I'm noting it as an edge case.